src/Neo/Wallets/Helper.cs

```diff
@@ -126,35 +128,76 @@ public static long CalculateNetworkFee(this Transaction tx, DataCache snapshot,
                     if (md.ReturnType != ContractParameterType.Boolean)
                         throw new ArgumentException("The verify method doesn't return boolean value.");
                     if (md.Parameters.Length > 0 && invocationScript is null)
-                        throw new ArgumentException("The verify method requires parameters that need to be passed via the witness' invocation script.");
+                    {
+                        var script = new ScriptBuilder();
+                        foreach (var par in md.Parameters)
+                        {
+                            switch (par.Type)
+                            {
+                                case ContractParameterType.Any:
+                                case ContractParameterType.Signature:
+                                case ContractParameterType.String:
+                                case ContractParameterType.ByteArray:
+                                    script.EmitPush(new byte[64]);
+                                    break;
+                                case ContractParameterType.Boolean:
+                                    script.EmitPush(true);
+                                    break;
+                                case ContractParameterType.Integer:
+                                    script.Emit(OpCode.PUSHINT256, new byte[Integer.MaxSize]);
+                                    break;
+                                case ContractParameterType.Hash160:
+                                    script.EmitPush(new byte[UInt160.Length]);
+                                    break;
+                                case ContractParameterType.Hash256:
+                                    script.EmitPush(new byte[UInt256.Length]);
+                                    break;
+                                case ContractParameterType.PublicKey:
+                                    script.EmitPush(new byte[33]);
+                                    break;
+                                case ContractParameterType.Array:
+                                    script.Emit(OpCode.PUSHINT256, new byte[Integer.MaxSize]);
+                                    script.Emit(OpCode.NEWARRAY);
+                                    break;
+                                case ContractParameterType.Map:
+                                    script.Emit(OpCode.NEWMAP);
```

Comment:
This provide no help to those kind of users. I don't think a contract with `Map` as it's `verify`'s parameter type would accept an empty map. So do the `Array`.

Maybe flowing to the default branch is better?

Reply:
@roman-khimov what's your opinion? This PR is for an issue opened by you.

Reply:
I don't think Map/Array should produce any code. We do our best in guessing the value, but it's hard to predict what does contract expect if it has Array or Map as an argument. That's why it's suggested not to push any code in these two cases. Likely, an error during verification will happen and thus, an error will be returned to the user, so that he'll understand that custom invocation script should be provided.

Reply:
Best to support this feature. Contract as a signer isn't widely support in `neo` core. Before `CalculateNetworkFee` only supported `contract` as signer with no arguments.